oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/web/DefaultBearerTokenResolverTests.java

```diff
@@ -128,6 +138,18 @@ public void resolveWhenFormParameterIsPresentAndSupportedThenTokenIsResolved() {
 		assertThat(this.resolver.resolve(request)).isEqualTo(TEST_TOKEN);
 	}
 
+	@Test
+	public void resolveWhenFormParameterIsPresentAndSupportedWithPutRequestThenTokenIsResolved() {
+		this.resolver.setAllowFormEncodedBodyParameter(true);
+
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.setMethod("PUT");
```

Comment:
Yes, I came across specs while doing this PR and realized the bug. This test case idea is to use cover something other than `GET` and `POST`. Thought about other HTTP methods but then it quickly becomes too many test cases with little value.

Reply:
If you consider how things are currently implemented then I understand it might appears that other tests for other methods add little value. However, since `POST`, `PUT`, `DELETE` and `PATCH` are the most common verbs used in HTTP based API it makes sense to add them as an insurance to prevent some future change inadvertently breaking those use cases.

Reply:
Added additional `DELETE` and `PATCH` test cases